dot/state/block.go

```diff
@@ -51,15 +51,17 @@ var (
 // BlockState contains the historical block data of the blockchain, including block headers and bodies.
 // It wraps the blocktree (which contains unfinalised blocks) and the database (which contains finalised blocks).
 type BlockState struct {
-	bt                *blocktree.BlockTree
-	baseState         *BaseState
-	dbPath            string
-	db                BlockStateDatabase
-	lock              sync.RWMutex
-	genesisHash       common.Hash
-	lastFinalised     common.Hash
-	unfinalisedBlocks *hashToBlockMap
-	tries             *Tries
+	bt                 *blocktree.BlockTree
+	baseState          *BaseState
+	dbPath             string
+	db                 BlockStateDatabase
+	lock               sync.RWMutex
+	genesisHash        common.Hash
+	lastFinalised      common.Hash
+	lastFinalisedRound uint64
+	lastFinalisedSetID uint64
```

Comment:
I would suggest using just `lastRound/lastSetID` since what was finalised was the block

```suggestion
	lastRound uint64
	lastSetID uint64
```